sys/net/crosslayer/ng_pktbuf_static/ng_pktbuf_static.c

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup net_ng_pktbuf
+ * @{
+ *
+ * @file
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/types.h>
+
+#include "mutex.h"
+#include "od.h"
+#include "utlist.h"
+#include "net/ng_pktbuf.h"
+#include "net/ng_nettype.h"
+#include "net/ng_pkt.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+typedef struct _unused {
+    struct _unused *next;
+    unsigned int size;
+} _unused_t;
+
+static mutex_t _mutex = MUTEX_INIT;
+static uint8_t _pktbuf[NG_PKTBUF_SIZE];
+static _unused_t *_first_unused = (_unused_t *)_pktbuf;
+
+/* internal ng_pktbuf functions */
+static ng_pktsnip_t *_create_snip(ng_pktsnip_t *next, void *data, size_t size,
+                                  ng_nettype_t type);
+static void *_pktbuf_alloc(size_t size);
+static void _pktbuf_free(void *data, size_t size);
+
+static inline bool _pktbuf_contains(void *ptr)
+{
+    return (&_pktbuf[0] <= (uint8_t *)ptr) &&
+           ((uint8_t *)ptr <= &_pktbuf[NG_PKTBUF_SIZE - 1]);
+}
+
+/* fits size to byte alignment */
+static inline size_t _align(size_t size)
+{
+    if (size % sizeof(void *)) {
```

Comment:
can this whole inline function be optimized to `size = (size + (sizeof(void *) -1)) & ~ (sizeof(void *) -1)` ?